src/iface/ethernet.rs

```diff
@@ -18,6 +18,7 @@ use socket::{Socket, SocketSet, AnySocket};
 #[cfg(feature = "proto-raw")] use socket::RawSocket;
 #[cfg(feature = "proto-udp")] use socket::UdpSocket;
 #[cfg(feature = "proto-tcp")] use socket::TcpSocket;
+#[cfg(feature = "proto-icmpv4")] use socket::Icmpv4Socket;
```

Comment:
Ugh. Just after I migrated `socket-*` features to `proto-*` I think we'll need to go back.

Reply:
Np. Should I add that in this PR?

Reply:
Sure.

Reply:
Updated